### DISCUSSION TITLE
Server becomes unstable when it reaches 1600 connections?

### DISCUSSION THREAD

**Discussion by SujithManjavana** [socketio/socket.io#4472] [Q&A]
I'm unable to get more than 1600 concurrent connections on my ubuntu server with Nginx reverse proxy. Detailed StackOverflow question can be found here => https://stackoverflow.com/questions/73757107/socket-io-scaling-issue

**Comment by darrachequesne** [comment]
Hi! There are several tips when scaling here https://socket.io/docs/v4/performance-tuning/#at-the-os-level, could it help?

**Reply by SujithManjavana** [reply]
HI @darrachequesne I have already read that article and installed `bufferutil utf-8-validate`. But it doesn’t change anything.

**Reply by SujithManjavana** [reply]
Could you please check that StackOverflow link?

**Reply by darrachequesne** [reply]
Let me try to reproduce the issue. Could it be related to the `worker_processes` option of nginx (which defaults to 1)?

Reference: https://www.nginx.com/resources/wiki/start/topics/examples/full/

**Reply by SujithManjavana** [reply]
This is my nginx.conf `user www-data;
worker_processes auto;
worker_rlimit_nofile 65535;
pid /run/nginx.pid;
include /etc/nginx/modules-enabled/*.conf;

events {
        worker_connections 30000;
        # multi_accept on;
}
...`

**Reply by SujithManjavana** [reply]
Also, I couldn’t find anything relevant in nginx error log.

**Reply by SujithManjavana** [reply]
@darrachequesne Please share your email ID. So that, I can share the full app code with you.

**Reply by qiulang** [reply]
@SujithManjavana  a better way is to create a private repo in github then invite someone to join, you may refer to this https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-access-to-your-personal-repositories/inviting-collaborators-to-a-personal-repository

**Reply by SujithManjavana** [reply]
@darrachequesne  UPDATE, I removed everything related to clustering and ran it again as a regular single-threaded nodejs app. This time, the result was a little better, 2800 stable connections.

**Reply by darrachequesne** [reply]
Unfortunately I wasn't able to reproduce. I have created a basic example with an Ansible playbook: https://github.com/socketio/socket.io-sample-playbook

On a t2.micro (1 vcpu, 1 Go RAM), I have just reached 10k clients without a sweat:

```
2022-09-26T08:48:24.588Z Worker 9447: 10058 connected client(s)
```

Could you please check?
